### PR TITLE
Feature to support dockerless compilation

### PR DESCRIPTION
**Pull request by AlvinHon**
It is merge request mainly for resolving issue (#6) which is to allow pchain_compile to compile contract without using docker.

Features:
- compile contract without using docker
- add command line option to specify the tag of docker image

Refactor: 
- provide struct Config as compilation configuration

### CODE REVIEW

File: src/manifests.rs [hunk 1]
```diff
@@ -51,12 +51,13 @@ pub fn package_name(current_dir: &Path) -> Result<String, Error> {
 /// Returns absolute path for a current directory.
 pub fn get_absolute_path(current_dir: &str) -> Result<String, Error> {
```
Comment (manngayin612) [inline src/manifests.rs:52]: Should we name the argument as "relative_path" or something other than "current_dir" to make it more generic?

so the comment can change to "Returns absolute path of a directory."

File: src/manifests.rs [hunk 2]
```diff
@@ -51,12 +51,13 @@ pub fn package_name(current_dir: &Path) -> Result<String, Error> {
...
     // get canonicalized path to the directory.
-    let canonicalized_path = dunce::canonicalize(current_dir)
-        .map_err(|_| Error::InvalidDependencyPath)?;
+    let canonicalized_path =
+        dunce::canonicalize(current_dir).map_err(|_| Error::InvalidDependencyPath)?;
 
     // also check if pchain-compile has write privileges to the canonicalized path.
     // if check passes, get absolute path to the directory.
```
Comment (manngayin612) [inline src/manifests.rs:58]: "**of** the directory" ?

**Review by manngayin612 — COMMENTED** [review]
There are just some minor fixes on comments.

File: src/bin/main.rs [hunk 3]
```diff
@@ -55,17 +73,33 @@ async fn main() {
...
             println!("Build process started. This could take several minutes for large contracts.");
 
+            let docker_option = if dockerless {
+                DockerOption::Dockerless
+            } else {
+                DockerOption::Docker(DockerConfig {
+                    tag: docker_image_tag,
+                })
+            };
+
```
[the remark this answers is not in the document]
Comment (AlvinHon) [reply src/bin/main.rs:92]: dockerless is optional field. If not specified, it is false by default.

docker_image_tag is optional field. If not specified, it uses a predefined tag by default.

So two options are serving different purposes, and they are mutually exclusive.

**Review by manngayin612 — APPROVED** [review]

**Pull request merged**